src/Window.cpp

```diff
@@ -1432,33 +1695,205 @@ bool Window::moveForward(int distance) {
     int moves = 1;
     while (moves < distance) {
         if (wallFront(moves)) {
-            break;
+            stopRun("Stop running because of a wall!");
+            return false;
```

Comment:
This is also an interesting change - allowing the simulator to end runs, rather than allowing the robot to repeatedly run into a wall. Can you convince me that this is the better behavior?